src/pages/projects/forms/NetworkSelector.tsx

```diff
@@ -0,0 +1,62 @@
+import { Select } from "@canonical/react-components";
+import type { SelectProps } from "@canonical/react-components";
+import type { FC } from "react";
+import { useNetworks } from "context/useNetworks";
+
+interface Props {
+  value: string;
+  project: string;
+  setValue: (value: string) => void;
+  selectProps?: SelectProps;
+  onBlur?: (e: React.FocusEvent) => void;
+  hasNoneOption?: boolean;
+}
+
+const NetworkSelector: FC<Props> = ({
+  value,
+  project,
+  setValue,
+  selectProps,
+  onBlur,
+  hasNoneOption = false,
+}) => {
```

Comment:
```suggestion
  setValue: (value: string) => void;
  onBlur?: (e: React.FocusEvent) => void;
  hasNoneOption?: boolean;
}

const NetworkSelector: FC<Props & SelectProps> = ({
  value,
  project,
  setValue,
  onBlur,
  hasNoneOption = false,
  ...selectProps
}) => {
```
I think if we inline the select props, the usage will be simpler, we don't need to nest the select props, and can directly use:

```
                      <NetworkSelector
                        ...
                        id={`devices.${index}.network`}
                        name={`devices.${index}.network`}
                        ...
```

Reply:
This took me a while to understand but I think I've got it now, please review the changes!